### PR TITLE
Add pre commit hook to run linter before each commit

### PR DESCRIPTION
**Pull request by RubenSousaDinis**
This PR adds a pre-commit hook that runs standardrb for staged files.

In order to setup the hook you will need to run the following command in the repo root folder:

`cp .githooks/pre-commit .git/hooks` 

Example output:

Screenshot 2022-01-27 at 21 16 46

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+lint: 
+	git diff --name-only --cached --diff-filter=d | grep '**/*\.rb\|**/*\.rake\|Gemfile' | grep -v 'Gemfile.lock' | xargs bundle exec standardrb --extra-details --fail-level A
```
Comment (RubenSousaDinis) [inline Makefile:2]: This is the command that runs the linter for staged files only.

File: .githooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
```
Comment (RubenSousaDinis) [inline .githooks/pre-commit:1]: This is the pre-commit script.

**Comment by pdfrod** [comment]
Nice! I just have a couple of comments:

1. There's still one pending warning when running the linter. Shouldn't we try to fix it, or at least add it to the TODO file? Screenshot_2022-01-27_22-30-24
2. What do you think of adding the linter instructions (i.e., `cp .githooks/pre-commit .git/hooks` and running `make lint`) to the README.md file?

**Comment by francisco-leal** [comment]
@pdfrod I think there is no way around it as Redis uses that global variable itself so I'm not sure we have an alternative. I'm sure there is a way to ignore however so that we don't get a warning

**Comment by francisco-leal** [comment]
@RubenSousaDinis I agree with @pdfrod, if we can add instructions to the README we'll be ready to ship this!

**Comment by RubenSousaDinis** [comment]
> @pdfrod I think there is no way around it as Redis uses that global variable itself so I'm not sure we have an alternative. I'm sure there is a way to ignore however so that we don't get a warning

I believe that the initializer can be removed. We're using heroku which means that we just need to point our app to the correct redis instance URL. Locally we can do the same and point to our local redis server.

**Comment by RubenSousaDinis** [comment]
> Nice! I just have a couple of comments:
> 
> 1. There's still one pending warning when running the linter. Shouldn't we try to fix it, or at least add it to the TODO file? Screenshot_2022-01-27_22-30-24
> 2. What do you think of adding the linter instructions (i.e., `cp .githooks/pre-commit .git/hooks` and running `make lint`) to the README.md file?

Readme notes added. @francisco-leal I typed a bit more about the git approach, feel free to change it.

**Review by verdefred — APPROVED** [review]
Pushed to our dev env and working great 🚀

**Pull request merged**